staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -992,7 +992,21 @@ func (e *Store) finalizeDelete(obj runtime.Object, runHooks bool) (runtime.Objec
 		}
 		return obj, nil
 	}
-	return &metav1.Status{Status: metav1.StatusSuccess}, nil
+	// Return information about the deleted object, which enables clients to
+	// verify that the object was actually deleted and not waiting for finalizers.
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+	gvk := obj.GetObjectKind().GroupVersionKind()
```

Comment:
This isn't going tone correct - you need to use the GVR passed to the store (Kind on StatusDetails really means Resource, this was a problem we created for ourselves)